libs/platform/src/lib/components/form/form-group/form-group.component.ts

```diff
@@ -282,56 +309,115 @@ export class FormGroupComponent implements FormGroupContainer, OnInit, AfterCont
         this.formGroup.removeControl(name);
     }
 
-    trackByFieldName(index: number, zoneField: GroupField): string | undefined {
-        return zoneField ? zoneField.name : undefined;
+    trackByFn(index: number): number  {
+        return index;
+    }
+
+    trackByFieldName(index: number, field: Field): string | undefined {
+        return field ? field.name : undefined;
+    }
+
+    isFieldGroupRow(row: KeyValue<FieldColumn, FieldGroup>): FieldColumn {
+        return row.value instanceof FieldGroup ? row.value.fields : row.value;
     }
 
     /**
-     * This reads FormFieldComponent from the QueyList and break all down into individual zones.
-     * Right now I support only 5 zones, but it could be easily extended to support any number of
-     * columns.
-     * Since I want to have it cached I know zones ahead of time, but you can have this generic and
-     * store everything in Map. Since here we have OnPush CD strategy it should not be a big deal.
+     * @hidden
+     * Assign a fields or field group to specified columns with rank.
+     * if `columnLayoutType` is given a fields without column property will set on last column.
+     * Otherwise the fields set 1 column.
      */
-    private updateFieldByZone(): void {
-        const zLeft: Array<GroupField> = [];
-        const zRight: Array<GroupField> = [];
-
-        this.formFieldChildren.forEach((item, index) => {
-            const zone: FormZone = this._multiLayout ? item.zone || 'zLeft' : 'zLeft';
-            const field = new GroupField(zone, item.id, item.rank || index, item.renderer, item.columns, item.fluid);
-
-            switch (zone) {
-                case 'zTop':
-                    if (!this.tZone) {
-                        this.tZone = [];
-                    }
-                    field.columns = 12;
-                    this.tZone.push(field);
-                    break;
-                case 'zBottom':
-                    if (!this.bZone) {
-                        this.bZone = [];
+    private _updateFieldByColumn(): void {
+        const rows: { [key: number]: FieldColumn } | { [key: number]: FieldGroup } = {};
```

Comment:
should not it will be easier? or this doesn't work?
```suggestion
        const rows: { [key: number]: FieldColumn | FieldGroup } = {};
```